maintenance: add command maintenance mode (#23)
red-hat-storage/odf-cli: README.md

## Commands

- `odf set`
  - `recovery-profile <profile>`: Set the recovery profile to favor new IO, recovery, or balanced mode with options `high_client_ops`, `high_recovery_ops`, or `balanced`. The default is `balanced`.
  - `ceph log-level <daemon> <subsystem> <log-level>`: Set the log level for Ceph daemons like OSD, mon, mds etc.
  More information about the ceph subsystems can be found [here](https://docs.ceph.com/en/latest/rados/troubleshooting/log-and-debug/#ceph-subsystems)
- `odf get`
  - `recovery-profile`: Get the recovery profile value.

> - `odf maintenance`: [Debug a deployment](docs/maintenance.md)  by scaling it down and creating a debug copy. This is supported for mons and OSDs only

Reviewer: ```suggestion
- `odf maintenance`: [Perform maintenance operations](docs/maintenance.md) on mons or OSDs. The mon or OSD deployment will be scaled down and replaced temporarily by a maintenance deployment.
```

  - `start <deployment-name>`

>     - `[--alternate-image <alternate-image>]` : Start debugging a deployment with an optional alternative ceph container image

Reviewer: ```suggestion
    - `[--alternate-image <alternate-image>]` : Start a maintenance deployment with an optional alternative ceph container image
```

>   - `stop <deployment-name>`: Stop debugging a deployment.

Reviewer: ```suggestion
  - `stop <deployment-name>`: Stop the maintenance deployment and restore the mon or OSD deployment
```

  - `recovery-profile`: Get the recovery profile value.
  - `health`: Check health of the cluster and common configuration issues.
- `odf purge-osd <ID>`: Permanently remove an OSD from the cluster.

> - `odf maintenance`: [Perform maintenance operations](docs/maintenance.md) on mons or OSDs. The mon or OSD deployment will be scaled down and replaced temporarily by a maintenance deployment.

Reviewer 2: maintenance is more general, how about troubleshoot?

Author: we already discussed and agreed with `maintenance` also the upstream PR is merged with `maintenance`. Sorry but you are a little late in the discussion:XD

  - `start <deployment-name>`

>     - `[--alternate-image <alternate-image>]` : Start a maintenance deployment with an optional alternative ceph container image

Reviewer 2: (optional) Start a maintenance deployment with an alternative ceph container image

Author: updated thanks

  - `stop <deployment-name>`: Stop the maintenance deployment and restore the mon or OSD deployment
- `odf subvolume`:
  - `ls`: Display all the subvolumes

>   - `delete <subvolume> <filesystem> <subvolumegroup>`: Deletes the stale subvolumes

Reviewer 2: <subvolume_name> <> similar

- `odf mons`
  - `restore-quorum <monID>`: Restore the mon quorum based on a single healthy mon since quorum was lost with the other mons

> - `odf help` : Display help text

Reviewer 2: displays all the odf cli commands with a help text

Author: it doesn't only display the command but flags, args, and root flags. So, it is standard to mention help text.
